rust/agama-lib/src/software/client.rs

```diff
@@ -20,7 +21,29 @@ pub struct Pattern {
     pub order: String,
 }
 
+/// Represents the reason why a pattern is selected.
+#[derive(Clone, Copy, Debug, Serialize)]
+pub enum SelectedBy {
+    /// The pattern was selected by the user.
+    User = 0,
+    /// The pattern was selected automatically.
+    Auto = 1,
+    /// The pattern has not be selected.
+    None = 2,
+}
+
+impl From<u8> for SelectedBy {
+    fn from(value: u8) -> Self {
+        match value {
+            0 => Self::User,
+            1 => Self::Auto,
+            _ => Self::None,
```

Comment:
I do not get this. Why on one side we convert "none" to 2 and here we accept anything? This looks kind of strange. I expect some kind of conversion error instead of silently accepting it.

Reply:
Yes, you are right, I will handle other cases.